flink-table-store-connector/src/test/java/org/apache/flink/table/store/connector/sink/LogStoreSinkITCase.java

```diff
@@ -0,0 +1,168 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.store.connector.sink;
+
+import org.apache.flink.api.common.eventtime.WatermarkStrategy;
+import org.apache.flink.streaming.api.datastream.DataStreamSource;
+import org.apache.flink.streaming.api.environment.StreamExecutionEnvironment;
+import org.apache.flink.table.data.RowData;
+import org.apache.flink.table.factories.DynamicTableFactory;
+import org.apache.flink.table.store.connector.FileStoreITCase;
+import org.apache.flink.table.store.file.FileStore;
+import org.apache.flink.table.store.kafka.KafkaLogSinkProvider;
+import org.apache.flink.table.store.kafka.KafkaLogSourceProvider;
+import org.apache.flink.table.store.kafka.KafkaLogStoreFactory;
+import org.apache.flink.table.store.kafka.KafkaLogTestUtils;
+import org.apache.flink.table.store.kafka.KafkaTableTestBase;
+import org.apache.flink.table.store.log.LogOptions;
+import org.apache.flink.types.Row;
+
+import org.junit.Test;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static org.apache.flink.table.store.connector.FileStoreITCase.buildBatchEnv;
+import static org.apache.flink.table.store.connector.FileStoreITCase.buildConfiguration;
+import static org.apache.flink.table.store.connector.FileStoreITCase.buildFileStore;
+import static org.apache.flink.table.store.connector.FileStoreITCase.buildStreamEnv;
+import static org.apache.flink.table.store.connector.FileStoreITCase.buildTestSource;
+import static org.apache.flink.table.store.kafka.KafkaLogTestUtils.SINK_CONTEXT;
+import static org.apache.flink.table.store.kafka.KafkaLogTestUtils.SOURCE_CONTEXT;
+import static org.apache.flink.table.store.kafka.KafkaLogTestUtils.discoverKafkaLogFactory;
+import static org.assertj.core.api.Assertions.assertThat;
+
+/** Test for table store with kafka. */
+public class LogStoreSinkITCase extends KafkaTableTestBase {
+
+    @Test
+    public void testStreamingPartitioned() throws Exception {
+        innerTest("testStreamingPartitioned", false, true, true);
+    }
+
+    @Test
+    public void testStreamingNonPartitioned() throws Exception {
+        innerTest("testStreamingNonPartitioned", false, false, true);
+    }
+
+    @Test
+    public void testBatchPartitioned() throws Exception {
+        innerTest("testBatchPartitioned", true, true, true);
+    }
+
+    @Test
+    public void testStreamingEventual() throws Exception {
+        innerTest("testStreamingEventual", false, true, false);
+    }
+
+    private void innerTest(String name, boolean isBatch, boolean partitioned, boolean transaction)
+            throws Exception {
+        StreamExecutionEnvironment env = isBatch ? buildBatchEnv() : buildStreamEnv();
+
+        // in eventual mode, failure will result in duplicate data
+        FileStore fileStore =
+                buildFileStore(
```

Comment:
It seems that `buildFileStore` always creates a file store with pk. Do we need to add the test that covers non-pk conditions?

Reply:
Yes, we can.
We need to refactor the FileStoreITCase, and even the Sink interface itself, which is a DataStream layer class that is more complex to build than a simple SQL can accomplish.
I create a jira, and subsequently we need to think through a problem, StoreSink exposed API should be what kind of, currently about keyed is rather confusing.

Reply:
https://issues.apache.org/jira/browse/FLINK-26530